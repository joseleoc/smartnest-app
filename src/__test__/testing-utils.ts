import GeneralContextProvider from "@/providers/generalContextProvider";
import { RenderOptions, render as rtlRender } from "@testing-library/react-native";
import { ReactElement } from "react";

function render<T>(component: ReactElement<T>, options?: RenderOptions) {
    return rtlRender(component, { wrapper: GeneralContextProvider, ...options });
}

export * from "@testing-library/react-native";

export { render }
