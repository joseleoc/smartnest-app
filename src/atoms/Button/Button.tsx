import React from "react";
import { useTheme } from "@react-navigation/native";
import { Pressable, PressableProps, StyleProp, ViewStyle } from "react-native";

import { ThemeProps } from "@/types";
import { styles } from "./Button.styles";

type ButtonProps = PressableProps &
  ThemeProps & { style?: StyleProp<ViewStyle>; pressStyle?: ViewStyle };

export default function Button(props: ButtonProps) {
  // --- Hooks -----------------------------------------------------------------
  const { colors } = useTheme();
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------

  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const { style, pressStyle, children } = props;
  // --- END: Data and handlers ------------------------------------------------

  return (
    <Pressable
      onPress={() => {}}
      style={({ pressed }) => [
        { backgroundColor: colors.primary },
        styles.buttonShadow,
        styles.button,
        pressed ? (pressStyle ?? style) : style,
      ]}
    >
      {children}
    </Pressable>
  );
}
