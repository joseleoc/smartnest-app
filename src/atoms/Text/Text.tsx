import useTheme from "@/hooks/useTheme";
import { ThemeProps } from "@/types";
import { Text as RNText, TextProps as RNTextProps } from "react-native";

export type TextProps = RNTextProps & ThemeProps;

export default function Text(props: TextProps) {
  // --- Hooks -----------------------------------------------------------------
  const {
    colors: { text },
  } = useTheme();
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const { style, lightColor, darkColor, children, ...otherProps } = props;
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  // --- END: Data and handlers ------------------------------------------------

  return (
    <RNText style={[{ color: text }, style]} {...otherProps}>
      {children}
    </RNText>
  );
}
