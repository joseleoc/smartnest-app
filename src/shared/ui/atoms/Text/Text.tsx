import useTheme from "@/hooks/useTheme";
import { ThemeProps } from "@/types";
import {
  Text as RNText,
  TextProps as RNTextProps,
  useColorScheme,
} from "react-native";
import { Theme } from "@/constants/theme";
import { useEffect, useState } from "react";

export type TextProps = RNTextProps &
  ThemeProps & { colorName?: keyof Theme["colors"] };

export default function Text(props: TextProps) {
  // --- Hooks -----------------------------------------------------------------
  const { colors } = useTheme();
  const colorScheme = useColorScheme();
  const [colorText, setColorText] = useState(colors.text);
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  const { style, lightColor, darkColor, children, colorName, ...otherProps } =
    props;

  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  useEffect(() => {
    if (colorName != null) {
      setColorText(colors[colorName]);
    }
  }, [colorScheme, colorName]);
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  // --- END: Data and handlers ------------------------------------------------

  return (
    <RNText style={[{ color: colorText }, style]} {...otherProps}>
      {children}
    </RNText>
  );
}
