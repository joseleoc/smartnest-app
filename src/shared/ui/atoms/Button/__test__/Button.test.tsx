import { render, screen, fireEvent } from "@/__test__/testing-utils";
import Button from "../Button";
import Text from "@/atoms/Text";

describe("Button atom", () => {
  test("renders correctly", () => {
    render(
      <Button>
        <Text>Test</Text>
      </Button>
    );
    expect(screen.queryByText("Test")).toBeTruthy();
  });

  test("fires onPress callback", () => {
    const onPress = jest.fn();
    render(
      <Button onPress={onPress}>
        <Text>Test</Text>
      </Button>
    );

    fireEvent.press(screen.queryByText("Test"));
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
