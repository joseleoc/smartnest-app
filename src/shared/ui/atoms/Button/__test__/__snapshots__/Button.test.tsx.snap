// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button atom renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#237050",
      },
      {
        "borderRadius": 8,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.25,
      },
      {
        "alignItems": "center",
        "borderRadius": 8,
        "justifyContent": "center",
        "paddingHorizontal": 24,
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;
